packages/firestore/src/api/field_value.ts

```diff
@@ -24,9 +24,23 @@ import {
 } from '../model/transform_operation';
 import { ParseContext, parseData, UserDataSource } from './user_data_reader';
 import { debugAssert } from '../util/assert';
-import { FieldValue } from '../../lite/src/api/field_value';
 import { toNumber } from '../remote/value_serializer';
 
+/**
+ * Sentinel values that can be used when writing document fields with `set()`
+ * or `update()`.
+ */
+export abstract class FieldValue {
```

Comment:
I can't say I really understand the layering model we have here and why this move is desirable.

It seems like conceptually the classic API should only ever point to the lite API and not the other way around. Allowing the other order encourages cycles and the possibility of polluting the lite implementation with classes that can't be tree shaken.

Reply:
Sorry, this is also something that was fixed in a later commit. It is indeed in lite/src/api/field_value